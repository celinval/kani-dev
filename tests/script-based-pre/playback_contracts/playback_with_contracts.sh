#!/usr/bin/env bash
# Copyright Kani Contributors
# SPDX-License-Identifier: Apache-2.0 OR MIT
# Test that concrete playback -Z concrete-playback executes as expected
set -o nounset

RS_FILE="modified.rs"
cp original.rs ${RS_FILE}

echo "[TEST] Generate test..."
kani ${RS_FILE} -Z concrete-playback --concrete-playback=inplace -Z function-contracts --output-format terse

# Note that today one of the tests will succeed since the contract pre-conditions are not inserted by Kani.
# Hopefully this will change with https://github.com/model-checking/kani/issues/3326
echo "[TEST] Run test..."
kani playback -Z concrete-playback ${RS_FILE} -- kani_concrete_playback

# Cleanup
rm ${RS_FILE}
rm kani_concrete_playback