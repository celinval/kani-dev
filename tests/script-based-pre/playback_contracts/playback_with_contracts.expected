[TEST] Generate test...
Checking harness check_modify_slice...

Failed Checks: | _ | slice[idx] == new_val
Failed Checks:\
in std::option::expect_failed

VERIFICATION:- FAILED

[TEST] Run test...

running 2 tests

Expected valid index, but contract is wrong
test result: FAILED. 1 passed; 1 failed;
